### DISCUSSION TITLE
String Lookup failed for name "lead_time"

### DISCUSSION THREAD

**Discussion by craigrj** [dtcenter/METplus#1797] [Configuration] [answered]
In my point stat configuration file, I have lead_time under the forecast block.  When MET runs, I get the above error.  What would cause that?

**Comment by JohnHalleyGotway** [comment, marked as answer]
@craigrj my guess would be that it's an issue with how the "lead_time" setting is formatted. 

For example, running this command:
```
plot_data_plane wrfprs_ruc13_12.tm00_G212 plot.ps 'name="TMP"; level="P500"; lead_time=12;'
```
Reproduces that warning message:
```
WARNING: Dictionary::lookup_string() -> string lookup failed for name "lead_time"
```
Note that the "lead_time" should be specified as a string, with double-quotes around. Rerunning that way, the warning message goes away:
```
plot_data_plane wrfprs_ruc13_12.tm00_G212 plot.ps 'name="TMP"; level="P500"; lead_time="12";'
```

Put simply, the lead_time should be defined as a string rather than a number.

Can you confirm that that was the source of the problem, and specifying it as a string solves it?

**Comment by craigrj** [comment]
That was it - Thanks

-----Original Message-----
From: johnhg ***@***.***> 
Sent: Monday, September 12, 2022 11:02 AM
To: dtcenter/METplus ***@***.***>
Cc: CRAIG, ROBERT J GS-12 USAF ACC 16 WS/WXD ***@***.***>; Mention ***@***.***>
Subject: [URL Verdict: Neutral][Non-DoD Source] Re: [dtcenter/METplus] String Lookup failed for name "lead_time" (Discussion #1797)

@craigrj <https://github.com/craigrj>  my guess would be that it's an issue with how the "lead_time" setting is formatted.

For example, running this command:

plot_data_plane wrfprs_ruc13_12.tm00_G212 plot.ps 'name="TMP"; level="P500"; lead_time=12;'


Reproduces that warning message:

WARNING: Dictionary::lookup_string() -> string lookup failed for name "lead_time"


Note that the "lead_time" should be specified as a string, with double-quotes around. Rerunning that way, the warning message goes away:

plot_data_plane wrfprs_ruc13_12.tm00_G212 plot.ps 'name="TMP"; level="P500"; lead_time="12";'


Put simply, the lead_time should be defined as a string rather than a number.

Can you confirm that that was the source of the problem, and specifying it as a string solves it?
